psi4/src/psi4/dfmp2/mp2.cc

```diff
@@ -1158,7 +1158,7 @@ void RDFMP2::form_Pab() {
 
     // Memory
     size_t doubles = ((size_t)(options_.get_double("DFMP2_MEM_FACTOR") * memory_ / 8L));
-    doubles -= navir * navir;
+    doubles -= static_cast<unsigned long> (navir) * navir;
```

Comment:
`navir` is an `int`. do we want to settle on `size_t` in this case?

Reply:
I think subtracting from `size_t` could lead either to undefined behavior or nasty surprises. Probably use a `long int`?

Reply:
@robertodr I don't think that helps here - it would have to be the type of `doubles` that gets changed to a signed quantity, right?  I think the best approach is to check that the size of `navir*navir` is less than doubles, before doing the subtraction, to avoid surprises.  Thoughts?

Reply:
I agree there should be a check that `navir*navir` isn't larger than `doubles`. Maybe `std::ptrdiff_t` is the appropriate type for `doubles` here? https://en.cppreference.com/w/cpp/types/ptrdiff_t

Reply:
Stepping back, oughtn't Wfn's `memory_` be a `size_t`?https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libmints/wavefunction.h#L120

Considering that Rob's checking below if the memory appears negative, rearranging types may be interfering.

Reply:
What change should be made to this code in the end? I had a hard time drawing a conclusion from this conversation.

Reply:
```C++
size_t navir2 = (size_t)navir * navir;
if ( (navir2) > double) { throw ...}

doubles -= navir2
```